### ISSUE TITLE
Name resolution failure causes crash

### ISSUE THREAD

**Issue by toresbe** [nais/herlighet#2]
A temporary DNS server inaccessibility revealed an issue where herlighet will crash if the target host cannot be resolved.

This should be handled better; a disconnect message should be returned to the client.

```
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Failed with result 'exit-code'.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Service RestartSec=100ms expired, scheduling restart.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Scheduled restart job, restart counter is at 11.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: Stopped herlighet.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: Started herlighet.
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]: 2021-04-20T01:53:59.480+0200        INFO        herlighet/posterior.go:61        Connecting as herlighet-static-readonly to b27dbvl023.preprod.local:5432/herlighet
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]: panic: dial tcp: lookup b27dbvl023.preprod.local: no such host
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]: goroutine 1 [running]:
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]: main.openDb(0x8dc100)
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]:         /home/runner/work/herlighet/herlighet/posterior.go:29 +0xcf
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]: main.smellTest()
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]:         /home/runner/work/herlighet/herlighet/posterior.go:36 +0x40
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]: main.main()
Apr 20 01:53:59 a30apvl042.oera.no herlighet[210005]:         /home/runner/work/herlighet/herlighet/main.go:31 +0x75
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Failed with result 'exit-code'.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Service RestartSec=100ms expired, scheduling restart.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Scheduled restart job, restart counter is at 12.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: Stopped herlighet.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Start request repeated too quickly.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: herlighet.service: Failed with result 'exit-code'.
Apr 20 01:53:59 a30apvl042.oera.no systemd[1]: Failed to start herlighet.
```

**Comment by toresbe** [comment]
This was actually a different issue than I expected. The panic is because the herlighet database itself is not reachable. There's no coming back from *that* problem, so I simply made the error message more informative.
